static/src/javascripts/projects/common/modules/commercial/dfp.js

```diff
@@ -404,30 +406,36 @@ define([
                     $placeholder.remove();
                     $adSlotContent.addClass('ad-slot__content');
                 });
-                checkForBreakout($slot);
-                addLabel($slot);
-                size = event.size.join(',');
-                // is there a callback for this size
-                if (callbacks[size]) {
-                    callbacks[size](event, $slot);
-                }
 
-                if ($slot.hasClass('ad-slot--container-inline') && $slot.hasClass('ad-slot--not-mobile')) {
-                    fastdom.write(function () {
-                        $slot.parent().css('display', 'flex');
-                    });
-                } else if (!($slot.hasClass('ad-slot--top-above-nav') && size === '1,1')) {
-                    fastdom.write(function () {
-                        $slot.parent().css('display', 'block');
-                    });
-                }
+                // Check if creative is a new gu style creative and place labels accordingly
+                checkForBreakout($slot).then(function (adType) {
+                    if (!adType || adType.type !== 'gu-style') {
```

Comment:
This file seems to be getting quite long and complicated, with lots of nesting and async behaviour. Should we maybe think about splitting it up somehow?